src/frontend/mame/ui/filesel.cpp

```diff
@@ -421,27 +420,29 @@ int menu_file_selector::compare_entries(const file_selector_entry *e1, const fil
 //-------------------------------------------------
 
 menu_file_selector::file_selector_entry *menu_file_selector::append_entry(
-	file_selector_entry_type entry_type, const char *entry_basename, const char *entry_fullpath)
+	file_selector_entry_type entry_type, const std::string &entry_basename, const std::string &entry_fullpath)
 {
-	file_selector_entry *entry;
-	file_selector_entry **entryptr;
+	return append_entry(entry_type, std::string(entry_basename), std::string(entry_fullpath));
+}
+
+
+//-------------------------------------------------
+//  append_entry - appends a new
+//  file selector entry to an entry list
+//-------------------------------------------------
 
+menu_file_selector::file_selector_entry *menu_file_selector::append_entry(
+	file_selector_entry_type entry_type, std::string &&entry_basename, std::string &&entry_fullpath)
+{
 	// allocate a new entry
-	entry = (file_selector_entry *) m_pool_alloc(sizeof(*entry));
-	memset(entry, 0, sizeof(*entry));
-	entry->type = entry_type;
-	entry->basename = (entry_basename != nullptr) ? pool_strdup(entry_basename) : entry_basename;
-	entry->fullpath = (entry_fullpath != nullptr) ? pool_strdup(entry_fullpath) : entry_fullpath;
+	file_selector_entry entry;
+	entry.type = entry_type;
+	entry.basename = std::move(entry_basename);
+	entry.fullpath = std::move(entry_fullpath);
 
 	// find the end of the list
-	entryptr = &m_entrylist;
-	while ((*entryptr != nullptr) && (compare_entries(entry, *entryptr) >= 0))
-		entryptr = &(*entryptr)->next;
-
-	// insert the entry
-	entry->next = *entryptr;
-	*entryptr = entry;
-	return entry;
+	m_entrylist.emplace_back(std::move(entry));
+	return &m_entrylist[m_entrylist.size() - 1];
```

Comment:
Since this never returns nullptr, why not make it return a reference rather than a pointer?